x-pack/plugins/ingest_manager/public/applications/ingest_manager/hooks/use_request/use_request.ts

```diff
@@ -35,3 +36,68 @@ export const useRequest = <D = any>(config: UseRequestConfig) => {
   }
   return _useRequest<D>(httpClient, config);
 };
+
+type SendConditionnalRequestConfig =
+  | (SendRequestConfig & { shouldSendRequest: true })
+  | (Partial<SendRequestConfig> & { shouldSendRequest: false });
+
+export const useConditionnalRequest = <D = any>(config: SendConditionnalRequestConfig) => {
+  const [state, setState] = useState<{
+    error: Error | null;
+    data: D | null;
+    isLoading: boolean;
+  }>({
+    error: null,
+    data: null,
+    isLoading: false,
+  });
+
+  const { path, method, shouldSendRequest, query, body } = config;
+
+  async function sendGetOneEnrollmentAPIKeyRequest() {
+    if (!config.shouldSendRequest) {
+      setState({
+        data: null,
+        isLoading: false,
+        error: null,
+      });
+      return;
+    }
+
+    try {
+      setState({
+        data: null,
+        isLoading: true,
+        error: null,
+      });
+      const res = await sendRequest<D>({
+        method: config.method,
+        path: config.path,
+        query: config.query,
+        body: config.body,
+      });
+      if (res.error) {
+        throw res.error;
+      }
+      setState({
+        data: res.data,
+        isLoading: false,
+        error: null,
+      });
+      return res;
+    } catch (error) {
+      setState({
+        data: null,
+        isLoading: false,
+        error,
+      });
+    }
+  }
+
+  useEffect(() => {
+    sendGetOneEnrollmentAPIKeyRequest();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
This is an important check. Why do we need to disable it?

Reply:
Because the hook depends on non scalar values `query` and `body` and I am using "stringifyed" version of these objects as dependencies.